src/main/java/com/indeed/jiraactions/ActionFactory.java

```diff
@@ -129,15 +167,63 @@ private long timeInState(final Action prevAction, final History history) {
     }
 
     private long timeInState(final Action prevAction, final DateTime changeTimestamp) {
-        if(!Objects.equals(prevAction.getPrevstatus(), prevAction.getStatus())) {
+        if (!Objects.equals(prevAction.getPrevstatus(), prevAction.getStatus())) {
             return getTimeDiff(prevAction.getTimestamp(), changeTimestamp);
         }
 
         return getTimeDiff(prevAction.getTimestamp(), changeTimestamp) + prevAction.getTimeinstate();
     }
 
+    private long getDateResolved(final String resolutionDate) {
+        if (resolutionDate == null || resolutionDate.equals("")) {
+            return 0;
+        } else {
+            return Long.parseLong(resolutionDate.split("T")[0].replaceAll("[^\\d]", ""));
+        }
+    }
+
+    private long getDateClosed(final Action prevAction, final History history) {
+        final String status = history.itemExist("status") ? history.getItemLastValue("status") : prevAction.getStatus();
+        if (status.equals("Closed")) {
+            if (prevAction.getStatus().equals(status)) {
+                return prevAction.getClosedDate();
+            }
+            return Integer.parseInt(history.created.toDateTimeISO().toString("yyyyMMdd"));
+        }
+        return 0;
+    }
+
     private long getTimeDiff(final DateTime before, final DateTime after) {
         return (after.getMillis() - before.getMillis()) / 1000;
     }
 
+    private List<String> createStatusHistory(final String status) {
+        final List<String> statusHistory = new ArrayList<>();
+        statusHistory.add(status);
+        return statusHistory;
+    }
+
+    private List<String> addStatusHistory(final List<String> prevHistory, final Action prevAction, final String status) {
+        final List<String> statusHistory = new ArrayList<>(prevHistory);
+        if (!status.equals(prevAction.getStatus())) {
+            statusHistory.add(status);
+        }
+        return statusHistory;
+    }
+
+    private long getDeliveryLeadTime(final Map<String, StatusTime> statusTimes, final Action action) {
+        if (action.getStatus().equals("Closed") &&
+        (DLT_RESOLUTIONS.contains(action.getResolution()) &&
+        (DLT_ISSUES.contains(action.getIssuetype())))) {
+            long deliveryLeadTime = 0;
+            for (final String status : dltStatuses) {
+                if (statusTimes.containsKey(status)) {
+                    deliveryLeadTime += statusTimes.get(status).getTimeinstatus();
+                }
+            }
+            return deliveryLeadTime;
+        }
+        return 0;
```

Comment:
I thought we had discussed previously about setting DLT as-is for issues that aren't closed.